### PR TITLE
Background clip

### LINKED ISSUE #196: Use correct `background-clip` where needed.
Use correct `background-clip` where needed, namely buttons, pickers etc.

### PR DESCRIPTION
**Pull request by joneff**
closes #196

### CODE REVIEW

**Comment by joneff** [comment]
@yordanov @lkarakoleva, can you list ALL instances where `background-clip: padding-box` is needed.

**Comment by yordanov** [comment]
### For kendo-angular-ui

Buttons
 - [x] Default Button
 - [x] ButtonGroup
 - [x] DropDownButton
 - [x] SplitButton

DateInputs
 - [x] Calendar wrapper
 
Dialog
 - [x] Dialog
 - [x] Dialog Buttons

DropDowns
 - [x] AutoComplete
 - [x] ComboBox
 - [x] DropDownList
 - [x] MultiSelect 
 - [x] MultiSelect Token

Grid
 - [x] Grid (.k-widget.k-grid)
 - [x] th & td

Inputs
 - [x] MaskedTextBox
 - [x] NumericTextBox
 - [x] Slider increase/decrease buttons
 - [x] Switch base + Switch handle

Layout
 - [x] PanelBar
 - [x] PanelBar root items
 - [x] TabStrip - content area
 - [x] TabStrip - active tab

Popup
 - [x] Popup

ScrollView
 - [x] .k-scrollview-wrap

Upload
 - [x] .k-dropzone
 - [x] .k-file item

**Comment by joneff** [comment]
@gyoshev @lkarakoleva @gyoshev @inikolova i think i've changed all (or at least most) instances. Have a look

**Comment by inikolova** [comment]
In IE 11 combination of background-clip: padding-box and border-radius causes additional white lines: 

https://github.com/telerik/kendo-theme-default/issues/479

**Comment by joneff** [comment]
The purest form of the snippet is:

```css
.issue {
    color: #fff;
    background-color: #000;
    border: 1px solid #000;
    background-clip: padding-box;
    border-radius: 1px;
}
```

Adding `vertical-align: middle` causes even more issues. Adding `display: inline-*` fixes some of them.

image

The last commit targets those by using scoped selectors for IE 11, Edge 12 and Edge 13 browsers.

File: scss/switch/_layout.scss [hunk 1]
```diff
@@ -98,5 +98,11 @@ $switch-handle-size: (.8 * $switch-size) !default;
         border-width: $switch-border-width;
         border-style: solid;
         background-clip: content-box;
+
```
Comment (inikolova) [inline scss/switch/_layout.scss]: Shouldn't this rule be in the _theme.scss file?

File: scss/toolbar/_layout.scss [hunk 2]
```diff
@@ -66,6 +66,12 @@
             position: absolute;
             top: 0;
             right: 0;
+
```
Comment (inikolova) [inline scss/toolbar/_layout.scss]: Shouldn't we move this rule to the _theme.scss file?

File: scss/menu/_layout.scss [hunk 3]
```diff
@@ -144,6 +144,12 @@
         background-clip: border-box;
         position: absolute;
 
+        .k-ie11 &,
```
Comment (inikolova) [inline scss/menu/_layout.scss]: Same as the comment above

**Review by inikolova — APPROVED** [review]

**Pull request merged**